pkg/golinters/goheader.go

```diff
@@ -15,16 +16,20 @@ import (
 
 const goHeaderName = "goheader"
 
-func NewGoHeader(settings *config.GoHeaderSettings) *goanalysis.Linter {
+func NewGoHeader(settings *config.GoHeaderSettings, cfg *config.Config) *goanalysis.Linter {
```

Comment:
1) Ambiguous API – you passed the whole config obj, but need the config dir only
2) Looks better to do overrides at `func (m Manager) GetAllSupportedLinterConfigs()` level
- will be consistent with other similar logics (like `GoVersion` update)
- will not require changing the constructor signature

Reply:
I agree, the proposed solution seems better.

[GoCritic](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/gocritic.go#L33) has the same config pattern though. Should that be changed as well?

Reply:
> [GoCritic](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/gocritic.go#L33) has the same config pattern though. Should that be changed as well?

Good point. But it is not relevant to your PR's goal, please do not add extra changes 👌